lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -39,7 +39,8 @@ class ConvertToImmutableTensors : public RewritePattern {
         opOperand.set(rewriter.create<CopyToValueTensorOp>(op->getLoc(),
                                                            opOperand.get()));
       } else if (auto listType = operandType.dyn_cast<ListType>()) {
-        if (!listType.getContainedType().isa<NonValueTensorType>())
+        if (!(listType.getContainedType().isa<NonValueTensorType>() ||
+              listType.getContainedType().isa<OptionalType>()))
```

Comment:
need to check the OptionalType is a `NonValueTensorType ` optional type.

Reply:
@cathyzhyi I have added a check to see that all tensors within optionalListType are non-value tensor types. Optional case handling is added as a TODO for now.